libraries/botframework-streaming/src/namedPipe/namedPipeServer.ts

```diff
@@ -70,36 +61,44 @@ export class NamedPipeServer implements IStreamingTransportServer {
      * @returns A promised string that will not resolve as long as the server is running.
      */
     public async start(): Promise<string> {
-        if (this._receiver.isConnected || this._sender.isConnected || this._incomingServer || this._outgoingServer) {
-            this.disconnect();
-        }
+        const { PipePath, ServerIncomingPath, ServerOutgoingPath } = NamedPipeTransport;
+
+        const incoming = new Promise<void>((resolve, reject) => {
+            const server = createNodeServer((socket) => {
+                if (this._receiver.isConnected) {

```

Comment:
This skips the connect logic for subsequent connections without terminating the underlying socket. My gut says it's bad to hold sockets open that aren't in use, but I don't know how Directline ASE/IISNode works with multiple Node procs and connections. A little wisdom here would be helpful!

Reply:
I like this! I think the previous implementation was an artifact of the original code being written by a C# engineer stumbling through writing a TypeScript library (me) and failing to make proper use of promises.